compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -160,9 +184,48 @@ OMR::ARM64::TreeEvaluator::imulhEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 TR::Register *
 OMR::ARM64::TreeEvaluator::lmulEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::lmulEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *src1Reg = cg->evaluate(firstChild);
+   TR::Register *src2Reg = NULL;
+   TR::Register *trgReg = NULL;
+
+   if(1 == firstChild->getReferenceCount())
+      {
+      trgReg = src1Reg;
+      }
+   else if(1 == secondChild->getReferenceCount() && secondChild->getRegister() != NULL)
+      {
+      trgReg = src2Reg;
+      }
+   else
+      {
+      trgReg = cg->allocateRegister();
+      }
+
+   if (secondChild->getOpCode().isLoadConst() && secondChild->getRegister() == NULL)
+      {
+      int64_t value = secondChild->getLongInt();
+      if (value > 0 && cg->convertMultiplyToShift(node))
+         {
+         trgReg = cg->evaluate(node);
```

Comment:
I think it is possible to reach here by taking the path through L203, which allocates a target register.  If that's the case, then that register will be lost.  Is it possible to move this logic where you decide to convert a multiply to a shift before the initial analysis starting at L193?  Alternatively, can you avoid allocating a target register if you know you'll be coming down this path?